File: android/plugins/plugin_example/src/main/java/github/tornaco/android/plugin/example/MainActivity.java
package github.tornaco.android.plugin.example;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.Nullable;

import github.tornaco.android.thanos.core.app.ThanosManager;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main);
        findViewById(R.id.button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ThanosManager thanos = ThanosManager.from(getApplicationContext());
                if (thanos.isServiceInstalled()) {
                    String fp = thanos.fingerPrint();
                    Toast.makeText(getApplicationContext(), "Hello world! \nThanox: " + fp, Toast.LENGTH_LONG).show();
                } else {
                    Toast.makeText(getApplicationContext(), "Hello world! \nThanox not active....", Toast.LENGTH_LONG).show();
                }
            }
        });
    }
}

File: android/sdkdemo/src/main/java/github/tornaco/thanox/sdk/demo/MainActivity.kt
package github.tornaco.thanox.sdk.demo

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import github.tornaco.android.thanos.theme.ThemeActivity
import github.tornaco.android.thanos.widget.ModernAlertDialog
import github.tornaco.thanox.sdk.demo.databinding.ActivityMainBinding

class MainActivity : ThemeActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setSupportActionBar(binding.toolbar)

        binding.fab.setOnClickListener {
            // Test modern dialog.
            ModernAlertDialog(this)
                .apply {
                    setDialogTitle("Android dev.")
                    setDialogMessage("Find the latest and greatest on the world's most powerful mobile platform. Browse devices, explore resources and learn about the latest updates.")
                    setPositive(getString(android.R.string.ok))
                    setNegative(getString(android.R.string.cancel))
                    setOnPositive {}
                    show()
                }.show()

        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return when (item.itemId) {
            R.id.action_settings -> true
            else -> super.onOptionsItemSelected(item)
        }
    }
}